### PR TITLE
LOTR 11 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Name:Press the Enemy
+ManaCost:2 U U
+Types:Instant
+A:SP$ ChangeZone | ValidTgts$ Permanent.nonLand,Card.inZoneStack | TgtZone$ Stack,Battlefield | Origin$ Battlefield,Stack | Fizzle$ True | Destination$ Hand | RememberChanged$ True | SubAbility$ DBMayPlay | SpellDescription$ Return target spell or nonland permanent an opponent controls to its owner's hand.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt:4]: Use `RememberLKI`

File: forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Name:Press the Enemy
+ManaCost:2 U U
+Types:Instant
+A:SP$ ChangeZone | ValidTgts$ Permanent.nonLand,Card.inZoneStack | TgtZone$ Stack,Battlefield | Origin$ Battlefield,Stack | Fizzle$ True | Destination$ Hand | RememberChanged$ True | SubAbility$ DBMayPlay | SpellDescription$ Return target spell or nonland permanent an opponent controls to its owner's hand.
+SVar:DBMayPlay:DB$ Play | Valid$ Card.YouOwn | ValidZone$ Hand| ValidSA$ Instant.cmcLEX,Sorcery.cmcLEX | WithoutManaCost$ True | Optional$ True | SubAbility$ DBCleanup | SpellDescription$ You may cast an instant or sorcery spell with equal or lesser mana value from your hand without paying its mana cost.
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+DeckHints:Type$Sorcery|Instant
+SVar:X:Targeted$CardManaCost
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt:8]: RememberedLKI$

File: forge-gui/res/cardsfolder/upcoming/rush_the_room.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Name:Rush the Room
+ManaCost:R
+Types:Instant
+A:SP$ Pump | ValidTgts$ Creature | TgtPrompt$ Select target creature | NumAtt$ +1 | SubAbility$ DBPump | SpellDescription$ Target creature gets +1/+0 and gains first strike until end of turn.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rush_the_room.txt:4]: first KW missing

File: forge-gui/res/cardsfolder/upcoming/rush_the_room.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Name:Rush the Room
+ManaCost:R
+Types:Instant
+A:SP$ Pump | ValidTgts$ Creature | TgtPrompt$ Select target creature | NumAtt$ +1 | SubAbility$ DBPump | SpellDescription$ Target creature gets +1/+0 and gains first strike until end of turn.
+SVar:DBPump:DB$ Pump | Defined$ Targeted | KW$ Haste | ConditionDefined$ Targeted | ConditionPresent$ Creature.Orc,Creature.Goblin | SpellDescription$ If it's a Goblin or Orc, it also gains haste until end of turn.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rush_the_room.txt:5]: Not creature

File: forge-gui/res/cardsfolder/upcoming/samwise_gamgee.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+Name:Samwise Gamgee
+ManaCost:G W
+Types:Legendary Creature Halfling Citizen
+PT:2/2
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Creature.nonToken+YouCtrl+Other | TriggerZones$ Battlefield | Execute$ TrigFood | TriggerDescription$ Whenever another nontoken creature enters the battlefield under your control, create a Food token. (It's an artifact with "{2}, Sacrifice this artifact: You gain 3 life.")
+SVar:TrigFood:DB$ Token | TokenAmount$ 1 | TokenScript$ c_a_food_sac | TokenOwner$ You | RememberTokens$ True | SubAbility$ DBImmediateTrig
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/samwise_gamgee.txt:6]: bad params

File: forge-gui/res/cardsfolder/upcoming/second_breakfast.txt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Name:Second Breakfast
+ManaCost:2 W
+Types:Instant
+A:SP$ Pump | TargetMin$ 0 | TargetMax$ 2 | NumAtt$ +2 | NumDef$ 1 | ValidTgts$ Creature | TgtPrompt$ Select target Creature | SubAbility$ DBFood | SpellDescription$ Up to two target creatures each get +2/+1 until end of turn.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/second_breakfast.txt:4]: avoid TgtPrompt for these simple types

File: forge-gui/res/cardsfolder/upcoming/shadow_of_the_enemy.txt [hunk 7]
```diff
@@ -0,0 +1,10 @@
+Name:Shadow of the Enemy
+ManaCost:3 B B B
+Types:Sorcery
+Oracle:
+A:SP$ ChangeZoneAll | Cost$ 2 B | Origin$ Graveyard | Destination$ Exile | ValidTgts$ Player | TgtPrompt$ Select target player | ChangeType$ Creature | RememberChanged$ True | SubAbility$ DBMayPlay | SpellDescription$ Exile all creature cards from target player's graveyard. You may cast spells from among those cards for as long as they remain exiled, and mana of any type can be spent to cast them.
+SVar:DBMayPlay:DB$ Effect | RememberObjects$ RememberedCard | StaticAbilities$ MayPlay | SubAbility$ DBCleanup | Duration$ Permanent | ForgetOnMoved$ Exile
+SVar:MayPlay:Mode$ Continuous | MayPlay$ True | MayPlayIgnoreColor$ True | EffectZone$ Command | Affected$ Card.IsRemembered | AffectedZone$ Exile | Description$ You may cast spells from among those cards for as long as they remain exiled, and mana of any type can be spent to cast them.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/shadow_of_the_enemy.txt:7]: - wrong Ignore param
- `+nonLand`

File: forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt]: The second part of the card works only when you return a nonland permanent on the board not on the stack

File: forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt [hunk 8]
```diff
@@ -0,0 +1,9 @@
+Name:Press the Enemy
+ManaCost:2 U U
+Types:Instant
+A:SP$ ChangeZone | ValidTgts$ Permanent.nonLand,Card.inZoneStack | TgtZone$ Stack,Battlefield | Origin$ Battlefield,Stack | Fizzle$ True | Destination$ Hand | RememberLKI$ True | SubAbility$ DBMayPlay | SpellDescription$ Return target spell or nonland permanent an opponent controls to its owner's hand.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt:4]: OppCtrl missing

File: forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt]: Try RememberChanged$ + `CardManaCostLKI`

Test with X spell as target, it should use the stack cmc

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt]: So I cast fireball for 6 and countered it with press the enemy but was not allowed to cast Time Warp (5 cmc) in my hand. Is this correct ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/press_the_enemy.txt]: not really I'll debug

**Review by tool4ever — APPROVED** [review]

**Pull request merged**